bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHAApplianceHandler.java

```diff
@@ -0,0 +1,426 @@
+                            }
+                            setApplianceProperties();
+                            updateStatus(ONLINE);
+                        }
+                    }
+                }
+            } catch (PlugwiseHAException e) {
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, STATUS_DESCRIPTION_COMMUNICATION_ERROR);
+            }
+        }
+    }
+
+    @Override
+    public void thingUpdated(Thing thing) {
+        super.thingUpdated(thing);
+
+        if (this.appliance.isBatteryOperated()) {
+            addBatteryChannels();
+        }
+
+        ThingHandler thingHandler = thing.getHandler();
+
+        if (thingHandler != null) {
+            for (Channel channel : thing.getChannels()) {
+                if (this.isLinked(channel.getUID())) {
+                    Appliance appliance = this.appliance;
+                    if (appliance != null) {
+                        this.refreshChannel(appliance, channel.getUID());
+                    }
+                }
+            }
+        }
+
+        setApplianceProperties();
+    }
+
+    @Override
+    protected @Nullable Appliance getEntity(PlugwiseHAController controller, Boolean forceRefresh)
+            throws PlugwiseHAException {
+        PlugwiseHAThingConfig config = getPlugwiseThingConfig();
+        Appliance appliance = controller.getAppliance(config.getId(), forceRefresh);
+
+        return appliance;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected void handleCommand(Appliance entity, ChannelUID channelUID, Command command) throws PlugwiseHAException {
+        String channelID = channelUID.getIdWithoutGroup();
+
+        PlugwiseHABridgeHandler bridge = this.getPlugwiseHABridge();
+        if (bridge == null) {
+            return;
+        }
+
+        PlugwiseHAController controller = bridge.getController();
+        if (controller == null) {
+            return;
+        }
+
+        switch (channelID) {
+            case APPLIANCE_LOCK_CHANNEL:
+                if (command instanceof OnOffType) {
+                    try {
+                        if (command == OnOffType.ON) {
+                            controller.switchRelayLockOn(entity);
+                        } else {
+                            controller.switchRelayLockOff(entity);
+                        }
+                    } catch (PlugwiseHAException e) {
+                        logger.warn("Unable to switch relay lock {} for appliance '{}'", (State) command,
+                                entity.getName());
+                    }
+                }
+                break;
+            case APPLIANCE_OFFSET_CHANNEL:
+                if (command instanceof QuantityType) {
+                    QuantityType<Temperature> state = (QuantityType<Temperature>) command;
+                    try {
+                        Unit<Temperature> unit = entity.getOffsetTemperatureUnit().orElse(UNIT_CELSIUS) == UNIT_CELSIUS
```

Comment:
Never compare strings with `==`. 
```suggestion
                        Unit<Temperature> unit = UNIT_CELSIUS.equals(entity.getOffsetTemperatureUnit().orElse(UNIT_CELSIUS))
```
Maybe also check other occurrences.

Reply:
Sorry and thanks, this is due to my lack of java experience. Learned something here ;-)